### PR TITLE
Don't rename the file if the replacement text is invalid

### PR DESCRIPTION
**Pull request by Cosifne**
Issue: https://github.com/dotnet/roslyn/issues/46208
A user could enter some invalid string by using copy & paste. And it causes a problem when trying to rename the file.
If the user renames a class and paste 'Class1<>' during the rename session, we will pass the value to System.IO.Path functions, then it will make things crash because '<' & '>' is not valid for a path.

The fix is to double-check if the replacement text valid before renaming the file.
I also make some UI changes so user would know it.

UI Changes:
1. When it starts renaming
image
2. If the text is not valid
image
3. If later user change the text to valid one.
image

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/InlineRename/Dashboard/DashboardViewModel.cs [hunk 1]
```diff
@@ -65,12 +66,36 @@ private void OnReferenceLocationsChanged(object sender, ImmutableArray<InlineRen
             UpdateSearchText(totalSpansCount, totalFilesCount);
         }
 
+        private void OnIsReplacementTextValidChanged(bool isReplacementTextValid)
```
Comment (ryzngard) [inline src/EditorFeatures/Core.Wpf/InlineRename/Dashboard/DashboardViewModel.cs:69]:
```suggestion
        private void OnIsReplacementTextValidChanged(bool isReplacementTextValid)
        {
            if (_isReplaceTextValid == isReplacementTextValid)
            {
                return;
            }
            
            _isReplacementTextValid = isReplacementTextValid;
            ComputeDefaultRenameFileFlag();
            NotifyPropertyChanged(nameof(AllowFileRename)); // Not sure if this is needed here, but we want to do NPC if the value is going to be computed differently
        }
        
        void ComputeDefaultRenameFileFlag()
        {
            DefaultRenameFileFlag = _isReplacementTextValid && 
                (_session.Option.GetOption(RenameOptions.RenameFile) || AllowFileRename);
        }
```

File: src/Workspaces/Core/Portable/Rename/ConflictEngine/MutableConflictResolution.cs [hunk 2]
```diff
@@ -89,7 +89,7 @@ internal void UpdateCurrentSolution(Solution solution)
                     var document = CurrentSolution.GetDocument(documentId);
                     var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
-                    // For the computeReplacementToken and computeReplacementNode functions, use 
+                    // For the computeReplacementToken and computeReplacementNode functions, use
```
Comment (ryzngard) [inline src/Workspaces/Core/Portable/Rename/ConflictEngine/MutableConflictResolution.cs:92]: Whitespace only change, please revert

File: src/EditorFeatures/Core.Wpf/InlineRename/Dashboard/DashboardViewModel.cs (hunk 1 above)
Comment (ryzngard) [reply src/EditorFeatures/Core.Wpf/InlineRename/Dashboard/DashboardViewModel.cs:69]: You can then call `ComputeDefaultRenameFileFlag()` in the constructor as well so the logic is the same in both places

File: src/Workspaces/Core/Portable/Rename/ConflictEngine/MutableConflictResolution.cs (hunk 2 above)
Comment (Cosifne) [reply src/Workspaces/Core/Portable/Rename/ConflictEngine/MutableConflictResolution.cs:92]: Looks like one of my plugin is tyring to do that. :) Will reset all the white space change

**Comment by Cosifne** [comment]
@ryzngard 
:) I have addressed all your comments.

**Review by ryzngard — APPROVED** [review]

**Pull request merged**